### PR TITLE
Use Java8 to build artifacts

### PR DESCRIPTION
**Pull request by jmatsu**
ref: https://github.com/DeployGate/gradle-deploygate-plugin/issues/152

Close https://github.com/DeployGate/gradle-deploygate-plugin/issues/165

This change drops AGP versions lower than 4.2.0 because of the compatibility. And also, Gradle is upgraded.

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -88,29 +88,13 @@ jobs:
       fail-fast: false
       matrix:
         agp_version:
-          - '3.5.1'
```
Comment (jmatsu) [inline .github/workflows/build-and-test.yml:91]: Dropped these AGP versions. They basically require Java7 binary.

File: example/app/build.gradle [hunk 2]
```diff
@@ -2,14 +2,13 @@ apply plugin: 'com.android.application'
 apply plugin: 'deploygate'
 
  android {
```
Comment (jmatsu) [inline example/app/build.gradle:4]: Updated the example project and remove unnecessary dependencies.

File: gradlew [hunk 3]
```diff
@@ -1,78 +1,129 @@
-#!/usr/bin/env sh
+#!/bin/sh
```
Comment (jmatsu) [inline gradlew:1]: Revised the wrapper file.

File: src/test/groovy/com/deploygate/gradle/plugins/tasks/factory/DeployGateTaskFactorySpec.groovy [hunk 4]
```diff
@@ -14,4 +14,15 @@ class DeployGateTaskFactorySpec extends Specification {
         "dep2"      | "uploadDeployGateDep2"
         "dep3Dep4"  | "uploadDeployGateDep3Dep4"
     }
+
+    def "uploadAabTaskName should return #taskName"() {
```
Comment (jmatsu) [inline src/test/groovy/com/deploygate/gradle/plugins/tasks/factory/DeployGateTaskFactorySpec.groovy:18]: missed this test :p

File: README.md [hunk 5]
```diff
@@ -94,6 +94,13 @@ See the test matrix of [.github/workflows/build-and-test.yml](./.github/workflow
 | 11  | 8.0.x-alpha06 (Flamingo) | 2.5.0 is OK              |
 | 17  | 8.0.x-alpha10 (Flamingo) | 2.5.0 is OK              |
 
+### Gradle compatibility
+
+| Gradle DeployGate Plugin | Minimum Gradle version |
+|:-------------------------|:-----------------------|
+| \<2.6                    | 5.4                    |
+ | \>=2.6.0                 | 6.7.1                  |
```
Comment (satsukies) [inline README.md:102]:
```suggestion
| \>=2.6.0                 | 6.7.1                  |
```

File: README_JP.md [hunk 6]
```diff
@@ -69,19 +69,19 @@ DeployGate Gradle プラグインは非 app モジュールや android-library 
...
-4.1.x | 2.3.0から
-4.2.x | 2.4.0から
-7.0.x (Arctic Fox) | 2.4.0から
-7.1.x (Bumblebee) | 2.4.0から
-7.2.x (Chipmunk) | 2.4.0から
-7.3.x (Dolphin) | 2.4.0から
-7.4.x (Electric Eel) | 2.4.0から
+| Android Gradle Plugin | Gradle DeployGate Plugin |
+|:----------------------|:-------------------------|
+| < 3.0.0               | 1.1.5 (保守終了)             |
```
Comment (satsukies) [inline README_JP.md:74]:
```suggestion
| < 3.0.0               | 1.1.5 (保守終了)           |
```

**Review by satsukies — APPROVED** [review]
LGTM.
It's just a nitpick, I post some comments. please check it.

**Comment by jmatsu** [comment]
Thank you for your reviews!

**Pull request merged**